### DISCUSSION TITLE
Mask are being ignored. I get car movements behind my mask.

### DISCUSSION THREAD

**Discussion by louisc32578** [blakeblackshear/frigate#3970] [Ask A Question]
With a mask covering above the horizontal line I still get motion actions. 

![Screenshot 2022-09-28 8 14 50 AM](https://user-images.githubusercontent.com/98428962/192788074-17d0420e-e1b1-4b42-82af-4e28754b67ac.png)

Driveway | car | 76.95% |   | 9/28/2022 | 7:50:05 AM | In progress

[Driveway](http://192.168.1.30:8123/events?camera=Driveway)	[car](http://192.168.1.30:8123/events?label=car)	76.95%	
9/28/2022	7:50:05 AM	In progress
![snapshot-1664369410 824648-82e3m0](https://user-images.githubusercontent.com/98428962/192788112-953a212a-80f9-4d30-bd45-17f515c99840.jpg)
![snapshot-1664368843 978433-hk6y2d](https://user-images.githubusercontent.com/98428962/192788202-ef648bce-df6f-4201-8550-2d437113b2bf.jpg)
![Screenshot 2022-09-28 8 17 17 AM](https://user-images.githubusercontent.com/98428962/192788821-1e177f9a-a0df-43f5-992f-112d27cb1666.png)

**Comment by NickM-27** [comment]
Please read the docs https://docs.frigate.video/configuration/masks

> Keep in mind that this mask only prevents motion from being detected and does not prevent objects from being detected if object detection was started due to motion in unmasked areas.

> Over masking will make it more difficult for objects to be tracked.

Motion masks are NOT meant to block off areas that are not of interest, that is what zones are for.

**Reply by NickM-27** [reply]
If you set `required_zones` then events from objects outside of the zone will not be saved

**Reply by louisc32578** [reply]
But will they be alarmed?  It appears shadows are causing the alarm.  So remove object detection.  This car has never driven into the unmasked area.

**Reply by NickM-27** [reply]
If you set `required_zones` then you will only get events when the car enters one of your zones. But the car is always tracked / detected. 

The point is that the motion masks will not stop the car in the neighbors driveway from being detected. It is also not meant to and motion masks are not meant to be used that way.

**Reply by louisc32578** [reply]
TY

**Comment by tlundste** [comment]
Thanks for the explanation on this... this helped me, as I was having the same problem (object detection in areas I didn't want it).  After reading the above, and documentation, I merely copied my motion mask area (an area where I cared nothing about, not motion, nor object detection), and pasted that into the
```
objects:
  filters:
    car:
      mask:
        - my motion mask numbers
        - my zone numbers for places cars shouldn't be
```
After that,  the debug display stopped drawing bounding boxes and identifying the cars in the driveways across the street.
Thanks!
